### PR TITLE
Math expressions in number textboxes

### PR DESCRIPTION
**Pull request by Dyvinia**
Allow typing math expressions in float/double textboxes. Feature that I'm personally a fan of since I use it in Blender but I'm unsure if it would fit in Frosty.

### CODE REVIEW

**Comment by wannkunstbeikor** [comment]
why only have this for floating points, and not integers?

**Comment by Dyvinia** [comment]
it doesn't seem as useful for ints but i can add it if you want

**Comment by wannkunstbeikor** [comment]
if u already add calculations it would only make sense to have it for all number types

**Comment by wannkunstbeikor** [comment]
i think if u can calculate in one field and not another it would be confusing, so it should work for all integer types so from byte to long and double and float

File: FrostyPlugin/Controls/Editors/FrostyNumberEditor.cs [hunk 1]
```diff
@@ -99,22 +125,63 @@ public override ValidationResult Validate(object value, CultureInfo cultureInfo)
...
+                            if (uint.TryParse(strValue, out uint _) || TrySolve(strValue, out double _))
+                            {
+                                return new ValidationResult(true, null);
+                            }
+                            else
+                            {
+                                return new ValidationResult(false, null);
+                            }
+                        }
                         catch (OverflowException) { return new ValidationResult(false, null); }
```
Comment (wannkunstbeikor) [inline FrostyPlugin/Controls/Editors/FrostyNumberEditor.cs]: those are also not needed if u use the tryparse right?

Comment (Dyvinia) [reply FrostyPlugin/Controls/Editors/FrostyNumberEditor.cs]: yea theyre not needed

**Pull request merged**